File: 121-BestTimeToBuyandSell/bestTimeToBuyandSell.java
public class Solution {
    public int maxProfit(int[] prices) {


        
        int rslt = 0;
        int min = Integer.MAX_VALUE;
        
        for (int i =0; i< prices.length; i++) {
            min = min < prices[i] ? min : prices[i];//min up to now
            int diff = prices[i] > min ? prices[i] -min : 0;
            rslt = rslt > diff ? rslt : diff;
        }

        

        
        return rslt;



        
    }
}

File: 121-BestTimeToBuyandSell/bestTimeToBuyAndSell.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.empty())  return 0;
        int prevMin=prices[0];
        int profit=0;
        for(int i=1;i<prices.size();i++){
            profit=max(profit,prices[i]-prevMin);
            prevMin=min(prevMin,prices[i]);
        }
        return profit;
    }
};